src/authentication/ldapjwtAuthProvider.tsx

```diff
@@ -0,0 +1,26 @@
+import Axios from 'axios';
+import JWTAuthProvider from './jwtAuthProvider';
+import { ScheduledMaintenanceState } from '../state/scigateway.types';
+import { MaintenanceState } from '../state/scigateway.types';
+
+export default class LDAPJWTAuthProvider extends JWTAuthProvider {
+  public fetchScheduledMaintenanceState(): Promise<ScheduledMaintenanceState> {
+    return Axios.get(`${this.authUrl}/scheduled_maintenance`)
+      .then((res) => {
+        return res.data;
+      })
+      .catch((err) => {
+        this.handleAuthError(err);
```

Comment:
The codecov warnings should be fixed, see `should log the user out if it fails to fetch scheduled maintenance state` in the icatAuth tests (same for the one below). I know you mentioned about whether it should do that but looking at it in more detail it only logs out when the error code is 403 or 401 so I think the test name is misleading but it also covers this line at the same time.